lib/lambda_ethereum_consensus/state_transition/accessors.ex

```diff
@@ -184,8 +184,13 @@ defmodule LambdaEthereumConsensus.StateTransition.Accessors do
   """
   @spec get_total_active_balance(BeaconState.t()) :: SszTypes.gwei()
   def get_total_active_balance(state) do
-    active_validator_indices = get_active_validator_indices(state, get_current_epoch(state))
-    get_total_balance(state, active_validator_indices)
+    epoch = get_current_epoch(state)
+
+    state.validators
+    |> Stream.filter(&Predicates.is_active_validator(&1, epoch))
+    |> Stream.map(fn %Validator{effective_balance: effective_balance} -> effective_balance end)
+    |> Enum.sum()
```

Comment:
is there any benefit in using stream here considering that you are going to do a Enum.sum afterwards?

Reply:
We would need benchmarks to know whether `Stream` or `Enum` is more efficient for this case, but at least it should consume less memory, as it can sum the result in batches, instead of first transforming the whole list and then reducing it.